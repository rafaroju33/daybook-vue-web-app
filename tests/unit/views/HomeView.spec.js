import HomeView from "@/views/HomeView.vue";
import { shallowMount } from "@vue/test-utils";

describe("Test HomeView component", () => {
  test("Should match Snapshot", () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("By clicking on a button should re-direct to no-entry", () => {
    const mockRouter = {
      push: jest.fn(),
    };

    const wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
