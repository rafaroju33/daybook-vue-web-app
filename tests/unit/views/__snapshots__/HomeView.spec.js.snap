// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomeView component Should match Snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
